src/components/tabs/ScatterPlotTab.tsx

```diff
@@ -0,0 +1,709 @@
+      if (!selectedTrack) {
+        return { xData: [], yData: [], objectIds: [], trackIds: [] };
+      }
+      for (let i = 0; i < selectedTrack.ids.length; i++) {
+        const id = selectedTrack.ids[i];
+        xData.push(rawXData[id]);
+        yData.push(rawYData[id]);
+      }
+      objectIds = Array.from(selectedTrack.ids);
+      trackIds = Array(selectedTrack.ids.length).fill(selectedTrack.trackId);
+    } else {
+      // All time
+      objectIds = [...Array(rawXData.length).keys()];
+      trackIds = Array.from(dataset!.trackIds || []);
+      // Copying the reference is faster than `Array.from()`.
+      xData = rawXData;
+      yData = rawYData;
+    }
+    return { xData, yData, objectIds, trackIds };
+  };
+
+  /**
+   * Creates the scatterplot and histogram axes for a given feature. Normalizes for dataset min/max to
+   * prevents axes from jumping during time or track playback.
+   * @param featureName Name of the feature to generate layouts for.
+   * @param histogramTrace The default histogram trace configuration.
+   * @returns An object with the following keys:
+   *  - `scatterPlotAxis`: Layout for the scatter plot axis.
+   *  - `histogramAxis`: Layout for the histogram axis.
+   *  - `histogramTrace`: A copy of the histogram trace, with potentially updated bin sizes.
+   */
+  const getAxisLayoutsFromRange = (
+    featureName: string,
+    histogramTrace: Partial<PlotData>
+  ): {
+    scatterPlotAxis: Partial<Plotly.LayoutAxis>;
+    histogramAxis: Partial<Plotly.LayoutAxis>;
+    histogramTrace: Partial<PlotData>;
+  } => {
+    let scatterPlotAxis: Partial<Plotly.LayoutAxis> = {
+      domain: [0, 0.8],
+      showgrid: false,
+      zeroline: true,
+    };
+    const histogramAxis: Partial<Plotly.LayoutAxis> = {
+      domain: [0.85, 1],
+      showgrid: false,
+      zeroline: true,
+      hoverformat: "f",
+    };
+    const newHistogramTrace = { ...histogramTrace };
+
+    let min = dataset?.getFeatureData(featureName)?.min || 0;
+    let max = dataset?.getFeatureData(featureName)?.max || 0;
+
+    // Special case for time feature, which isn't in the dataset
+    if (featureName === TIME_FEATURE.name) {
+      min = 0;
+      max = dataset?.numberOfFrames || 0;
+    }
+
+    if (dataset && dataset.isFeatureCategorical(featureName)) {
+      // Add extra padding for categories so they're nicely centered
+      min -= 0.5;
+      max += 0.5;
+    } else {
+      // Add a little padding to the min/max so points aren't cut off by the edge of the plot.
+      // (ideally this would be a pixel padding, but plotly doesn't support that.)
+      min -= (max - min) / 10;
+      max += (max - min) / 10;
+    }
+    scatterPlotAxis.range = [min, max];
+
+    // TODO: Show categories as box and whisker plots instead of scatterplot?
+    // TODO: Add special handling for integer features once implemented, so their histograms use reasonable
+    // bin sizes to prevent jumping.
```

Comment:
could these TODOs be turned into Issues?

Reply:
The integer handling is an issue: https://github.com/allen-cell-animated/nucmorph-colorizer/issues/163
The categories one currently isn't, but needs more input from scientists. I'll write up an issue ticket for it.

Reply:
https://github.com/allen-cell-animated/nucmorph-colorizer/issues/192